### DISCUSSION TITLE
how to decode inconsistent json

### DISCUSSION THREAD

**Discussion by vajrabisj** [vlang/v#18901] [Questions and Answers] [answered]
the json is like following:
[{"description":"something","category":"home","done":[]}, {"description":"other thing","category":"office","done":true}]

the value to "done" is inconsistent, including [], true and false

i just confused on this, and don't know how to decode it. To use x.json2? there is a Any type. But how? thanks in advance.

**Comment by vajrabisj** [comment, marked as answer]
resolved.

**Reply by JalonSolov** [reply]
How did you resolve it?  Do you have a small example?  Perhaps it could be added to the documentation for future users.

**Reply by vajrabisj** [reply]
Actually I found json.decode is smart to parse these inconsistent by just declare the done as bool in the struct :), after decoding, []/false turn to false, true still be true

**Comment by tzSharing** [comment]
Maybe try "sum type".
For example: 
`type doneValue = [] string | bool`
